src/vs/workbench/contrib/sessionSync/browser/sessionSync.contribution.ts

```diff
@@ -0,0 +1,179 @@
+		@IProgressService private readonly progressService: IProgressService,
+		@ISCMService private readonly scmService: ISCMService,
+		@IConfigurationService private configurationService: IConfigurationService,
+		@IWorkspaceContextService private readonly contextService: IWorkspaceContextService,
+	) {
+		super();
+
+		this.configurationService.onDidChangeConfiguration((e) => {
+			if (e.affectsConfiguration('workbench.experimental.sessionSync.enabled')) {
+				this.registerActions();
+			}
+		});
+
+		this.registerActions();
+	}
+
+	private registerActions() {
+		if (this.registered || this.configurationService.getValue('workbench.experimental.sessionSync.enabled') === false) {
+			return;
+		}
+
+		this.registerApplyEditSessionAction();
+		this.registerStoreEditSessionAction();
+
+		this.registered = true;
+	}
+
+	private registerApplyEditSessionAction(): void {
+		const that = this;
+		this._register(registerAction2(class ApplyEditSessionAction extends Action2 {
+			constructor() {
+				super({
+					id: applyLatestCommand.id,
+					title: applyLatestCommand.title,
+					menu: {
+						id: MenuId.CommandPalette,
+					}
+				});
+			}
+
+			async run(accessor: ServicesAccessor): Promise<void> {
+				await that.progressService.withProgress({
+					location: ProgressLocation.Notification,
+					title: localize('applying edit session', 'Applying edit session...')
+				}, async () => await that.applyEditSession());
+			}
+		}));
+	}
+
+	private registerStoreEditSessionAction(): void {
+		const that = this;
+		this._register(registerAction2(class StoreEditSessionAction extends Action2 {
+			constructor() {
+				super({
+					id: storeLatestCommand.id,
+					title: storeLatestCommand.title,
+					menu: {
+						id: MenuId.CommandPalette,
+					}
+				});
+			}
+
+			async run(accessor: ServicesAccessor): Promise<void> {
+				await that.progressService.withProgress({
+					location: ProgressLocation.Notification,
+					title: localize('storing edit session', 'Storing edit session...')
+				}, async () => await that.storeEditSession());
+			}
+		}));
+	}
+
+	async applyEditSession() {
+		const editSession = await this.sessionSyncWorkbenchService.read();
+		if (!editSession) {
+			return;
+		}
+
+		for (const folder of editSession.folders) {
+			const folderRoot = this.contextService.getWorkspace().folders.find((f) => f.name === folder.name);
+			if (!folderRoot) {
+				return;
+			}
+
+			for (const { relativeFilePath, contents, type } of folder.workingChanges) {
+				const uri = joinPath(folderRoot.uri, relativeFilePath);
+				if (type === ChangeType.Addition) {
+					await this.fileService.writeFile(uri, VSBuffer.fromString(contents));
+				} else if (type === ChangeType.Deletion && await this.fileService.exists(uri)) {
+					await this.fileService.del(uri);
+				}
+			}
+		}
+	}
+
+	async storeEditSession() {
+		const folders: Folder[] = [];
+
+		for (const repository of this.scmService.repositories) {
```

Comment:
@lszomoru, I believe you own the SCM service, and I'm interested in your feedback on whether it's advisable for the Continue Edit Session functionality to depend on the SCM service. In brief, for Continue Edit Session to work, we need to know which files contain uncommitted changes. I would like to avoid introducing new API for this unless absolutely necessary. The SCM service does have knowledge of which files have changed, they are the resources tracked in the SCM providers' resource groups. I'm not sure if this violates any layering rules in the VS Code codebase, so wanted to check with you on this.

Reply:
This seems reasonable to me but I am adding @joaomoreno to make sure I am not missing something.

Reply:
I think this makes sense. But I'd love to know more as to why

> for Continue Edit Session to work, we need to know which files contain uncommitted changes.

Reply:
Makes sense! LGTM!